src/jit/MemoryInl.h

```diff
@@ -233,6 +263,9 @@ static void emitLoad(sljit_compiler* compiler, Instruction* instr)
         opcode = (instr->info() & Instruction::kHasFloatOperand) ? SLJIT_MOV_F64 : SLJIT_MOV;
         size = 8;
         break;
+#if defined(ENABLE_EXTENDED_FEATURES)
+    case ByteCode::I32AtomicLoadOpcode:
```

Comment:
This should set `CheckNaturalAlignment`.

I would create an `options = 0` variable at the beginning of the function and overwrite it in these cases.